### DISCUSSION TITLE
Is there a way to accomplish new data arrived behavior?

### DISCUSSION THREAD

**Discussion by FranciscoGileno** [TanStack/query#1069] [General]
1. Query is executed for the first time, request is made, data is saved on cache and displayed.
2. Same query is executed, new data is requested from the server.
3. If the data is different from previous, then alert the user with an action: "New data arrived, click here to update".

OBS: I read the docs, but didn't find anything.

**Comment by TkDodo** [comment]
So once the data hits the query cache, it is in there and everyone subscriber will just receive that data, so I don't think possible out of the box.

A quite stupid idea: implement a custom `isDataEqual` function and keep a reference to `previousData` and `data` passed to it (possibly with a useRef or a useState), and then always return `true`. Because you return `true`, the `previousData` will stay in the cache (see [here](https://github.com/tannerlinsley/react-query/blob/45371c9571d83e86268ee2b7e68c2cd100e80af2/src/core/query.ts#L205-L208)). You would need an exception for the first query (`previousData === undefined` or so). Then, you can do the equal comparison manually and display your alert to the user. If the user chooses `update`, you put the `newData` into the cache manually via `queryCache.setData`. Not sure if that really works though, haven't tried it. Might be stupid, but might also be genius, it's a thin line to walk 😅 .
